app/controllers/application_controller.rb

```diff
@@ -3,5 +3,19 @@
 # :nocov:
 # Application controller
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:email, :password, :password_confirmation, :current_password, :location)
```

Comment:
You have to permit nested parameters here :) like `user_locations_attributes` and `location_attributes`

Reply:
@ag4ta and @mpapis  you guys also suggested writing a test for this method
How do I do that?

Reply:
I think request specs for creating and updating user should be enough. hmm :)

Additionally we could set `action_on_unpermitted_parameters` to raise exception in `development` and `test` environments. What do you think?

Reply:
https://edgeguides.rubyonrails.org/action_controller_overview.html#nested-parameters 

@ag4ta nice job on `action_on_unpermitted_parameters`, I did not know about it!  It's time for your first commit to rails guides: https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters actually: https://github.com/rails/rails/blob/master/guides/source/action_controller_overview.md#strong-parameters

@ponyczek you can find more details here: https://api.rubyonrails.org/classes/ActionController/Parameters.html